### DISCUSSION TITLE
[Craft 5] Option to stack element cards

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#14202] [Ideas] [answered]
The `card view` for elements is a great addition and I look forward to using it a lot in my projects. A thing that bugs me is that in many cases I would prefer to view the cards in a single column, not in a grid. This is especially true in the case of a classic Matrix "content builder", but also in the Entry Index view (depending on the type of section)

Reasons the grid style view can become problematic:

1. The order of blocks/entries becomes fuzzy/harder to parse. Should I read that as rows? Or maybe as columns? See example below:

![image](https://github.com/craftcms/cms/assets/19857599/201f0ddf-bf1c-4796-9bfe-43c43d1acb68)

The order in a single column layout is much clearer:

![image](https://github.com/craftcms/cms/assets/19857599/133546c3-30a9-4d17-bef0-30f3a9f21c6e)

2. A grid could be mistaken by the author to imply a specific layout on the front-end: Cards that happen to appear side-by-side in the CP could be read as a "row", whereas a single column just shows elements one after the other.

There are of course cases where the grid style is preferable, so I would propose to have an option for the display of cards (`grid` or `stacked` ) in the context of a matrix field, and also in the context of an element index.

Thanks for considering this!

**Comment by wmdhosting** [comment]
I tought that too.. :)

Perhaps just add setting  when choosing  that option. similar to  field..  width

**Comment by alexanderloewe** [comment]
+1

**Comment by brandonkelly** [comment, marked as answer]
We discussed this internally and the consensus was that we should just always show cards in full-width regardless of context or screen size. It’s clearly easier to reorder them when in single-column, and even in places where they aren’t sortable, it’s good to be consistent.

Craft 5.0.0-alpha.9 was just tagged with that change.

**Reply by wsydney76** [reply]
Hm, not sure, obviously there are different use cases for that card view.

This is what an entries index with card view will look like:

![image](https://github.com/craftcms/cms/assets/60010597/24f25b70-224f-4cbb-8006-da7add408005)

A lot of empty space, a lot of scrolling required for the editor, this now loses the advantage of a clear, compact view (apart from a larger image).

So an option to choose between stacked and grid layout would be highly appreciated.

/W.

PS,
Tbh: Requires courage to approach a client, saying 'Hey, we have better view with better AX now!' and then present this...

**Reply by khalwat** [reply]
Yeah, I concur; this does not look good at all.

**Reply by ryanirelan** [reply]
This is disappointing. Now, the card view offers no real advantage over the table view. They're both now suffering from the same full-width view problem. 

![CleanShot 2024-01-30 at 11 26 06@2x](https://github.com/craftcms/cms/assets/2238/58081677-8268-4025-90db-675ca62b2cd5)

Look at the line length! Please reconsider this change to a much-touted new display feature in Craft 5.

**Reply by brandonkelly** [reply]
Would you folks agree full-width is better in cases where the cards are sortable? Or do you prefer the grid view even there?

**Reply by wmdhosting** [reply]
This is somethign needed for sure,  but previus too :)

**Reply by wsydney76** [reply]
Looks like a grid layout with draggable cardlets like the one below are an established pattern in other systems. We've been using this for a long time now and haven't heard of a single problem with it. 

![image](https://github.com/craftcms/cms/assets/60010597/f4fb75f7-82c4-4584-a146-d9a612d802fb)

At least cards should have a sensible maximum width.

**Reply by mmikkel** [reply]
I share @thupsi's concerns re: the grid view; particularly for "page builder" use cases or the like, due to the implied layout and ambiguous order. If a field setting to choose between grid and full-width view is off the table, personally I'd prefer the full-width view as the safer option.

I do agree that the cards currently look a bit awkward in full-width on wider screens, but that can probably be improved?

**Reply by brandonkelly** [reply]
> We've been using this for a long time now and haven't heard of a single problem with it.

Good to hear. I personally didn’t find it difficult either, but assumed I was alone in that due to being the guy that built it, since everyone else was saying they didn’t like it (here, internally, and in person at Dot All).

It sounds like we should add a new “Limit card views to single-column” user preference that determines the behavior across all card views, rather than making it a field setting/index setting.

**Reply by lindseydiloreto** [reply]
My two cents... both modes should be possible. In fact it shouldn't even be different "modes" for card view, the developer should simply be able to choose the **number of columns** (as low as one).

And both modes (there should be only one "mode") should be sortable. If the developer finds that multiple columns are confusing for a client, they could simply **reduce the number of columns to `1`** for that particular project.

**Reply by lindseydiloreto** [reply]
@brandonkelly Love that idea, we kinda said the same thing at the same time. 😄 It's a "number of columns" issue, nothing more.

**Reply by brandonkelly** [reply]
> the developer should simply be able to choose the number of columns (as low as one)

That seems awkward since ideally the number of columns is responsive based on the container width. I think it should be a binary choice.

**Reply by lindseydiloreto** [reply]
> the number of columns is responsive based on the container width

I'll take your word for it, that's outside of my wheelhouse. I'm mostly making the point that it's a column number issue, and everything else about the previous card system was great already.

**Reply by brandonkelly** [reply]
> I share @thupsi's concerns re: the grid view; particularly for "page builder" use cases or the like, due to the implied layout and ambiguous order.

That’s a good point. Maybe it does just come down to the use case, and thus makes sense to be a field setting.

**Reply by mmikkel** [reply]
> That’s a good point. Maybe it does just come down to the use case, and thus makes sense to be a field setting.

Exactly. I don't actually think the AX w/ sortable, multi-column element cards is fundamentally different from, say, "Large thumbnail" asset fields in Craft 4, and nobody has ever had a problem with those. _I.e. for simple "related elements" type fields, the multi-column grid view is easily the most compact and effective._

For page builder-like content, however – where each element _typically_ represents a full-width component on the front end – multiple columns will likely provide a much more confusing AX, for the reasons outlined by @thupsi.

A field setting is the perfect solve IMO. In general I don't think authors are necessarily thrown off by inconsistency across similar types of input; some UI variation is completely OK if it makes a thing more intuitive _in the context of the specific content being edited_.

**Reply by brandonkelly** [reply]
Alright, I [brought back](https://github.com/craftcms/cms/commit/278aef8f460a74dd96e5147fd44698a5c694ec7c) multi-column card grids, and added a new “Show cards in a grid” setting to Matrix fields (disabled by default), for the next release.

<img src="https://github.com/craftcms/cms/assets/47792/6d0bf2b3-9571-4b1d-95cb-99de4c889c8e" width="518" alt="A Matrix field’s “View Mode” setting set to “As cards”, and “Show cards in a grid” setting, which is enabled.">

**Reply by thupsi** [reply]
Wow, this conversation exploded and I missed it! Pretty happy with the result though 🙂

**Reply by brandonkelly** [reply]
5.0.0-alpha.10 is out with those changes. Thanks for all the feedback, everyone ❤️

**Reply by brandonkelly** [reply]
Just released Craft 5.0.0-alpha.12 with a “Show cards in a grid” setting added to relational fields, too.
